pyblish_maya/lib.py

```diff
@@ -313,3 +310,28 @@ def _show_no_gui():
 
     messagebox.setStandardButtons(messagebox.Ok)
     messagebox.exec_()
+
+
+class Dock(QtWidgets.QWidget):
+
+    def __init__(self, dock_count, parent=None):
+        super(Dock, self).__init__(parent)
+        QtWidgets.QVBoxLayout(self)
+        object_name = "pyblish_maya.dock" + str(dock_count)
+        self.setObjectName(object_name)
+
+
+def dock(window):
+
+    main_window = None
+    for obj in QtWidgets.qApp.topLevelWidgets():
+        if obj.objectName() == "MayaWindow":
+            main_window = obj
+
+    self._dock_count += 1
```

Comment:
That's a good question.

It's possible we could have `show()` instantiate a window once, and then reuse it automatically, by just holding onto the previous window internally.

If a widget was shown this way, `window.show()`, how does Docking respond to that?